Webinf autocompleter 177548593 (#298)
mavenlink/design-system: src/components/autocompleter/autocompleter.md

The `Autocompleter` component is a wrapper for Select component with the additional functionality searching for a specific models from an API endpoint.

> `apiEndpoint`, should be the remainder of the apiEndpoint (excluding the base api), eg. `/workspaces`.

Reviewer: ```suggestion
 `apiEndpoint` should be the route of the api's endpoint (excluding the base api), eg. `/workspaces`.
```

`value` and `models` shape is expected to be an object(s) with an `id` key, and a display attribute that is one of the following: `title`, `name`, `full_name`, `currency`;

> For a11z, and additional prop info, see the `Select` component.

Reviewer: ```suggestion
For a11y and additional prop info, see the `Select` component.
```

Author: lol

### Default examples
```js
import Autocompleter from './autocompleter.jsx';

> <Autocompleter apiEndpoint={'/models'} />

Reviewer: It might be good to show a real-ish example here like `<Autocompleter apiEndpoint="workspaces" />`. I think models is pretty clear but using a model that Mavenlink actually has could cement the idea.

Author: I kiiiiinda like the abstract example. My intent was to show that it works for anything. Is that more clear with a specific example 🤷‍♂️

Reviewer: Works for me! I think it's nit-picky feedback anyway

The `Autocompleter` component is a wrapper for Select component with the additional functionality searching for a specific models from an API endpoint.

> `apiEndpoint` should be the route of the api's endpoint (excluding the base api), eg. `/workspaces`.

Reviewer 2: This can be inline documentation so it shows up in the props table. e.g. https://github.com/mavenlink/design-system/blob/ac91df1426e45289eaf10bb985ec8359bc6b5e57/src/components/date/date.jsx#L238

> `value` and `models` shape is expected to be an object(s) with an `id` key, and a display attribute that is one of the following: `title`, `name`, `full_name`, `currency`;

Reviewer 2: If you describe the prop type in `Autocompleter.propTypes` then this would be auto-documented in the props table. 

I think you'll still need to add some text for the display attribute. But maybe it's possible to describe that as prop-types too. I'm not sure though.

> For a11y and additional prop info, see the `Select` component.

Reviewer 2: ```suggestion
For a11y and additional prop info, see the `Select` component.

```

### Default examples
```js
import Autocompleter from './autocompleter.jsx';

> <Autocompleter apiEndpoint={'/models'} id='ex-1' name='default-ex' label='Default Example' />

Reviewer 2: ```suggestion
<Autocompleter apiEndpoint='/models' id='ex-1' name='default-ex' label='Default Example' />
```

```

### Initial value examples
```js
import Autocompleter from './autocompleter.jsx';

> <Autocompleter apiEndpoint={'/models'} value={{id: 11, name: 'cool'}} id='ex-2' name='value-ex' label='Value Example' />

Reviewer 2: ```suggestion
<Autocompleter apiEndpoint='/models' value={{id: 11, name: 'cool'}} id='ex-2' name='value-ex' label='Value Example' />
```

```

## Ref API Example
```jsx
import Autocompleter from './autocompleter.jsx';
import RefExample from '@mavenlink/design-system/src/components/ref-example/ref-example.jsx';

const ref = React.createRef();

<RefExample ref={ref}>
  {({ onChange }) => (

>     <Autocompleter ref={ref} onChange={onChange} apiEndpoint={'/models'} id='ex-3' name='ref-ex' label='Ref Example'  />

Reviewer 2: ```suggestion
    <Autocompleter ref={ref} onChange={onChange} apiEndpoint='/models' id='ex-3' name='ref-ex' label='Ref Example'  />
```
